src/main/java/baritone/pathing/movement/movements/MovementParkourAdv.java

```diff
@@ -0,0 +1,1054 @@
+                    lowestCost = getMoveResult(context, destX, srcY - decendAmount, destZ, extraAscend, VecUtils.add(posbJump, 0, -decendAmount, 0), simpleDirection, type, 0, lowestCost, res);
+                    continue;
+                    //}
+                    //System.out.println("TEST 4, Flat Failed");
+                    //continue;
+                }
+            }
+
+            //No block to land on, we now test for a parkour place
+
+            if (!MovementHelper.fullyPassable(context, destX, srcY + 3, destZ)) {
+                continue; //If block above head at destination
+            }
+
+            if (!context.allowParkourPlace) {
+                continue; // Settings don't allow a parkour place
+            }
+
+            //System.out.println("TEST 6, Parkour Place, in dir " + simpleDirection);
+
+            // time 2 pop off with that dank skynet parkour place
+            IBlockState toReplace = context.get(destX, srcY - 1, destZ);
+            double placeCost = context.costOfPlacingAt(destX, srcY - 1, destZ, toReplace);
+            if (placeCost >= COST_INF) {
+                continue; // Not allowed to place here
+            }
+            if (!MovementHelper.isReplaceable(destX, srcY - 1, destZ, toReplace, context.bsi)) {
+                continue; // Not allowed to place here
+            }
+            //TODO
+            //if (!checkOvershootSafety(context.bsi, destX + xDiff, srcY, destZ + zDiff)) {
+                //continue; // Can place but will overshoot into a bad location
+            //}
+
+            //Check if a block side is available/visible to place on
+            for (int j = 0; j < 5; j++) {
+                int againstX = destX + HORIZONTALS_BUT_ALSO_DOWN_____SO_EVERY_DIRECTION_EXCEPT_UP[j].getXOffset();
+                int againstY = srcY - 1 + HORIZONTALS_BUT_ALSO_DOWN_____SO_EVERY_DIRECTION_EXCEPT_UP[j].getYOffset();
+                int againstZ = destZ + HORIZONTALS_BUT_ALSO_DOWN_____SO_EVERY_DIRECTION_EXCEPT_UP[j].getZOffset();
+                if (againstX == srcX + xDiff * 3 && againstZ == srcZ + zDiff * 3) { // we can't turn around that fast
+                    continue;
+                }
+                if (MovementHelper.canPlaceAgainst(context.bsi, againstX, againstY, againstZ)) {
+                    lowestCost = getMoveResult(context, destX, srcY, destZ, extraAscend, posbJump, simpleDirection, type, placeCost, lowestCost, res);
+                    continue;
+                }
+            }
+        }
+        res = lowestCost;
+        if (res.cost < 1000) {
+            Vec3i jumpVec = new Vec3i(res.x - srcX, res.y - srcY, res.z - srcZ);
+            //System.out.println("Dir = " + simpleDirection + ", Cost: " + res.cost + ", Distance: " + getDistance(jumpVec, simpleDirection) + ", MoveDis: " + calcMoveDist(context, srcX, srcY, srcZ, extraAscend, jumpVec, simpleDirection));
```

Comment:
I suggest installing a linter

Reply:
How do linter's (first time I've heard of them sorry) help here?
(I was logging each jump so that I could see what the maximum jump was)

Reply:
A linter tells you when you write bad code

Reply:
> A linter tells you when you write bad code

So, I'd just get red lines or something on all my code? :)